### DISCUSSION TITLE
Custom RGBA value formatting error

### DISCUSSION THREAD

**Discussion by dy** [tailwindlabs/tailwindcss#10141] [Help]
There's friction with some bundlers like esbuild https://github.com/evanw/esbuild/issues/2763.
As potential workaround we could use `bg-[rgba(193_195_201/0.7)]` but tailwind doesn't recognize custom bg value in that format.

tailwindcss@3.0.16
esbuild@0.16.10

**Comment by adamwathan** [comment]
Hey! I agree with Evan in the esbuild thread that you should be processing with Tailwind before running through esbuild.

Re: your workaround, it should work if you use the right function — needs to be `rgb` not `rgba`:

```
bg-[rgb(193_195_201/0.7)]
```

Either way unfortunately this isn't something we can do anything about in Tailwind — esbuild is the tool that is changing the code here and at the same time I think it's totally reasonable that esbuild expects only standard CSS syntax as input.

Going to convert this to a discussion, maybe you can share more about how your project is set up and I can give you some more feedback on what I would do to just avoid this situation altogether.

**Reply by dy** [reply]
Thank you for the reply.
There turns to be some friction integrating component libraries that use tailwind in its internals.
There doesn't seem to be apparent best-practice or guide on that.

We have 100+ components looking like `**/*.tsx` and `**/*.css`, where `tsx` imports css as `import './index.css'`.
The lib is built by standard rollup config, internally similar to what esbuild does: just bundling tsx/css together.
That forces consumers of this library to include full bundle, which is unnecessary weight from both CSS/JS.
We attempted bundle each component individually via solution from [this discussion](https://github.com/tailwindlabs/tailwindcss/discussions/6612#discussioncomment-4430041) (since tailwind doesn't do multiple files), but tailwind for each individual component produces many utility classes, overlapping other components.
